scipy/stats/_continuous_distns.py

```diff
@@ -4242,6 +4247,53 @@ def _munp(self, n):
     def _entropy(self):
         return 2-np.log(2)
 
+    @_call_super_mom
+    @inherit_docstring_from(rv_continuous)
+    def fit(self, data, *args, **kwds):
+        if (isinstance(data, CensoredData)) or kwds.pop('superfit', False):
+            return super().fit(data, *args, **kwds)
+
+        data, floc, fscale = _check_fit_input_parameters(self, data,
+                                                         args, kwds)
+
+        if floc is not None or fscale is not None:
+            return super().fit(data, *args, **kwds)
+
+        # location is the minimum of the sample ([1] Equation 2.3)
+        loc = np.min(data)
+
+        # scale is solution to a fix point problem ([1] 2.6)
+        # use approximate MLE as starting point ([1] 3.1)
+        n_observations = data.shape[0]
+        sorted_data = np.sort(data, axis=0)
+        p = np.arange(0, n_observations, 1)/(n_observations + 1)
+        q = 1 - p
+        pp1 = 1 + p
+        alpha = p - 0.5 * q * pp1 * np.log(pp1 / q)
+        beta = 0.5 * q * pp1
+        sorted_data = sorted_data - loc
+        B = 2 * np.sum(alpha[1:] * sorted_data[1:])
+        C = 2 * np.sum(beta[1:] * sorted_data[1:]**2)
+        # starting guess
+        scale = ((B + np.sqrt(B**2 + 8 * n_observations * C))
+                 /(4 * n_observations))
+
+        # relative tolerance of fix point iterator
+        rtol = 1e-8
+        relative_residual = 1
+        shifted_mean = sorted_data.mean() #y_mean - y_min
```

Comment:
```suggestion
        shifted_mean = sorted_data.mean()  # y_mean - y_min
```
https://peps.python.org/pep-0008/#inline-comments